app/src/main/java/com/eveningoutpost/dexdrip/wearintegration/WatchBroadcast/WatchBroadcastService.java

```diff
@@ -0,0 +1,535 @@
+
+        JoH.startService(WatchBroadcastService.class, INTENT_FUNCTION_KEY, CMD_START);
+        super.onCreate();
+    }
+
+    @Override
+    public void onDestroy() {
+        UserError.Log.e(TAG, "killing service");
+        broadcastEntities.clear();
+        unregisterReceiver(broadcastReceiver);
+        super.onDestroy();
+    }
+
+    @Override
+    public int onStartCommand(Intent intent, int flags, int startId) {
+        final PowerManager.WakeLock wl = JoH.getWakeLock(TAG, 10000);
+        try {
+            if (isEnabled()) {
+                if (intent != null) {
+                    final String function = intent.getStringExtra(INTENT_FUNCTION_KEY);
+                    if (function != null) {
+                        try {
+                            handleCommand(function, intent);
+                        } catch (Exception e) {
+                            UserError.Log.e(TAG, "handleCommand Error: " + e.toString());
+                        }
+                    } else {
+                        UserError.Log.d(TAG, "onStartCommand called without function");
+                    }
+                }
+                return START_STICKY;
+            } else {
+                UserError.Log.d(TAG, "Service is NOT set be active - shutting down");
+                stopSelf();
+                return START_NOT_STICKY;
+            }
+        } finally {
+            JoH.releaseWakeLock(wl);
+        }
+    }
+
+    private void handleCommand(String function, Intent intent) {
+        UserError.Log.d(TAG, "handleCommand function:" + function);
+        String receiver = null;
+        boolean handled = false;
+        WatchBroadcast watchBroadcast;
+        Bundle bundle = null;
+        //send to all connected apps
+        for (Map.Entry<String, WatchBroadcast> entry : broadcastEntities.entrySet()) {
+            receiver = entry.getKey();
+            watchBroadcast = entry.getValue();
+            switch (function) {
+                case CMD_UPDATE_BG:
+                    bundle = prepareBgBundle(watchBroadcast);
+                    sendBroadcast(function, receiver, bundle);
+                    break;
+            }
+        }
+        if (function.equals(CMD_UPDATE_BG)) {
+            handled = true;
+        }
+
+        if (handled) {
+            return;
+        }
+        switch (function) {
+            case CMD_REPLY_MSG:
+                bundle = new Bundle();
+                bundle.putString(INTENT_REPLY_MSG, intent.getStringExtra(INTENT_REPLY_MSG));
+                bundle.putString(INTENT_REPLY_CODE, intent.getStringExtra(INTENT_REPLY_CODE));
+                receiver = intent.getStringExtra(INTENT_PACKAGE_KEY);
+                break;
+            case CMD_ALERT:
+                bundle = new Bundle();
+                bundle.putString("type", intent.getStringExtra("type"));
+                bundle.putString("message", intent.getStringExtra("message"));
+                break;
+            case CMD_START:
```

Comment:
no, it will send a broadcast message with function CMD_START to all receivers. This is useful to notify other applications about xdrip service start so they can handle this message and register themselves as a receiver